src/backends/cuda/cuda_device.cpp

```diff
@@ -113,15 +110,21 @@ namespace luisa::compute::cuda {
                               luisa::to_underlying(format));
 }
 
-CUDADevice::CUDADevice(Context &&ctx,
-                       size_t device_id,
-                       const BinaryIO *io) noexcept
-    : DeviceInterface{std::move(ctx)}, _handle{device_id}, _io{io} {
+CUDADevice::CUDADevice(
+    Context &&ctx,
+    size_t device_id,
+    const BinaryIO *io
+) noexcept
+    : DeviceInterface { std::move(ctx) }
+    , _handle { device_id }
+    , _default_io { luisa::make_unique<DefaultBinaryIO>(context()) }
+    , _io { io }
+{
     // provide a default binary IO
     if (_io == nullptr) {
-        _default_io = luisa::make_unique<DefaultBinaryIO>(context());
```

Comment:
This change is incorrect. The original implementation only creates `_default_io` when input `io` is nullptr, but now it's always created.